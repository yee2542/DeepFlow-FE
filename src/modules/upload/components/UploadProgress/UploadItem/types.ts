import { GroupTask } from 'Modules/upload/model/types';

type UploadItemProps = {
  file: {
    file: GroupTask;
  };
};

type ProgressBarPropsType = {
  width: number;
};

export type { UploadItemProps, ProgressBarPropsType };
