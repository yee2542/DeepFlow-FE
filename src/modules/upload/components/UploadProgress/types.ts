import { GroupTask } from '../../model/types';

export type UploadProgressPropsType = {
  group: GroupTask[];
  total: number;
};
