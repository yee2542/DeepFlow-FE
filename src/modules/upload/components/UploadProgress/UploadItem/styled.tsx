import styled from 'styled-components';
import { ProgressBarPropsType } from './types';

export const WrapperItem = styled.div`
  display: grid;
  grid-template-columns: calc(100% - 50px) 50px;
  padding: 16px;
  box-shadow: 0px 1px 2px lightgray;
`;

export const LeftSide = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-right: 8px;

  label {
    width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;
    text-align: left;
  }
`;

export const ProgressBarWrapper = styled.div`
  width: 100%;
  height: 15px;
  background-color: lightgray;
  margin-bottom: 8px;
  border-radius: 20px;

  div {
    height: 15px;
    background-color: lightgreen;
    border-radius: 20px;
  }
`;

export const Percentage = styled.span`
  margin-left: 12px;
`;

export const ProgressBar = styled.div<ProgressBarPropsType>`
  ${(props) =>
    props.width &&
    `
   width: ${props.width}%`}
`;
