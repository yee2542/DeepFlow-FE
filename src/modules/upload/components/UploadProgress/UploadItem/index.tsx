import React from 'react';
import { WrapperItem, LeftSide, ProgressBar, Percentage } from './styled';
import { UploadItemProps } from './types';

const UploadItem: React.FC<UploadItemProps> = (props) => {
  const { file } = props.file;
  return file ? (
    <WrapperItem>
      <LeftSide>
        <ProgressBar>
          <div style={{ width: `${file.progress}%` }} />
          <label>{file.label}</label>
        </ProgressBar>
      </LeftSide>
      <Percentage>{file.progress}</Percentage>
    </WrapperItem>
  ) : null;
};

export default UploadItem;
