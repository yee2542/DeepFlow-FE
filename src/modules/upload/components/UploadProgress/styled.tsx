import styled from 'styled-components';

export const Wrapper = styled.div`
  position: fixed;
  bottom: 0;
  right: 0;
  background-color: white;
  width: 400px;
  overflow: scroll;
  max-height: 400px;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
  h5 {
    background-color: var(--secondary-color);
    margin: 0;
    color: white;
    padding: 20px;
  }
`;
