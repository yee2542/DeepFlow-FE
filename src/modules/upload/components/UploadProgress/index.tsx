import React from 'react';
import UploadItem from './UploadItem';
import { useSelector } from 'react-redux';
import { StoresState } from 'Stores/index';
import { Wrapper } from './styled';
import { v4 as uuid } from 'uuid';
const UploadProgress: React.FC = () => {
  const group = useSelector((state: StoresState) => state.upload.present.group);
  const total = useSelector((state: StoresState) => state.upload.present.total);

  return total > 0 ? (
    <Wrapper>
      <h4>Uploading File</h4>
      {total > 0
        ? Object.values(group).map((file) => {
            return <UploadItem key={uuid()} file={{ file: file }} />;
          })
        : null}
    </Wrapper>
  ) : null;
};

export default UploadProgress;
