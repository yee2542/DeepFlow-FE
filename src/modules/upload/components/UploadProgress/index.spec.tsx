import UploadProgress from '.';
import renderer from 'react-test-renderer';
import React from 'react';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import UploadItem from './UploadItem';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';

const mockStore = configureMockStore([thunk]);

const mockGroupState = [
  { label: 'individual1', progress: 1, total: 1 },
  { label: 'group1', progress: 4, total: 10 },
];

const mockTotalState = 2;

describe('Components/UploadProgress test ', () => {
  const store = mockStore({
    upload: {
      present: {
        group: mockGroupState,
        total: mockTotalState,
      },
    },
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('Should match snapshot', () => {
    const wrap = renderer
      .create(
        <Provider store={store}>
          <UploadProgress />
        </Provider>,
      )
      .toJSON();
    expect(wrap).toMatchSnapshot();
  });

  it('total > 0 should render upload item', () => {
    const wrap = mount(
      <Provider store={store}>
        <UploadProgress />
      </Provider>,
    );
    expect(wrap.find(UploadItem)).toHaveLength(2);
  });
});
