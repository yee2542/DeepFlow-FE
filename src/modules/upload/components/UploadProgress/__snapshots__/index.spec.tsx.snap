// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/UploadProgress test  Should match snapshot 1`] = `
<div
  className="styled__Wrapper-hutygj-0 kOrSMU"
>
  <h4>
    Uploading File
  </h4>
  <div
    className="styled__WrapperItem-ukzgv1-0 efQqci"
  >
    <div
      className="styled__LeftSide-ukzgv1-1 jWWiFz"
    >
      <div
        className="styled__ProgressBarWrapper-ukzgv1-2 gPOYAe"
      >
        <div
          className="styled__ProgressBar-ukzgv1-4 bvhUjA"
          width={1}
        />
        <label>
          individual1
        </label>
      </div>
    </div>
    <span
      className="styled__Percentage-ukzgv1-3 lojdNb"
    >
      1
    </span>
  </div>
  <div
    className="styled__WrapperItem-ukzgv1-0 efQqci"
  >
    <div
      className="styled__LeftSide-ukzgv1-1 jWWiFz"
    >
      <div
        className="styled__ProgressBarWrapper-ukzgv1-2 gPOYAe"
      >
        <div
          className="styled__ProgressBar-ukzgv1-4 SsSLN"
          width={4}
        />
        <label>
          group1
        </label>
      </div>
    </div>
    <span
      className="styled__Percentage-ukzgv1-3 lojdNb"
    >
      4
    </span>
  </div>
</div>
`;
